src/foam/nanos/u2/navigation/SideNavigation.js

```diff
@@ -0,0 +1,147 @@
+foam.CLASS({
+    package: 'foam.nanos.u2.navigation',
+    name: 'SideNavigation',
+    extends: 'foam.u2.View',
+  
+    documentation: 'Side navigation bar',
+  
+    implements: [
+        'foam.mlang.Expressions'
+      ],
+       requires: [
+        'foam.nanos.menu.Menu'
+      ],
+       imports: [
+        'menuDAO'
+      ],
+       css: `
+        ^ .side-nav {
+          height: 100%;
+          width: 200px;
+          position
+          z-index: 1;
+          top: 0;
+          left: 0;
+          overflow-x: hidden;
+          background-color: white;
+          display: inline-block;
+        }
+        ^ .nav-row {
+          display: block;
+        }
+        ^ .side-nav a {
+          display: inline-block;
+          margin: 8px 8px 8px 8px;
+          text-decoration: none;
+          font-size: 20px;
+          transition: 0.3s;
+        }
+        ^ .side-nav a:hover {
+          color: #f1f1f1;
+        }
+        ^ .accordion-card a {
+          font-size: 16px;
+          margin: 8px 8px 8px 50px;
+        }
+        ^ .accordion-card-hide {
+          display: none;
+        }
+        ^ .accordion-card-show {
+          display: block !important;
+        }
+        ^ .accordion-button {
+          display: inline-block;
+          border: none;
+          padding: 0;
+          margin: 8px 8px;
+          font-size: 20px;
+          overflow: hidden;
+          text-decoration: none;
+          text-align: left;
+          cursor: pointer;
+          white-space: nowrap;
+        }
+        ^ .accordion-button:focus {
+          outline: 0;
+        }
+      `,
+       properties: [
+        {
+          name: 'accordionCardShowDict',
+          value: {}
+        },
+        {
+          class: 'Boolean',
+          name: 'accordionCardShow',
+          value: true
+        },
+        {
+          class: 'String',
+          name: 'menuName',
+        }
+      ],
+       methods: [
+        function initE() {
```

Comment:
I think this view is trying to do too much, and is causing some limitations and complication as a result.

This view appears to be responsible for rendering the entire tree of menu items.  As a result, it only appears to support one level of nesting of menu items, and the click handler for those menu items is repeated twice.  Additionally, state is being maintained imperatively in a property (accordingShowCardDict) rather than declaratively all the way down.

A simpler design would be to write a recursive view that is responsible for rendering a single menu item, and expanding/collapsing the children for that menu item.  The children will be rendered using another instance of the same view.  That way if if the child menus have their own children, we can expand and collapse those for free as well.

Here's some example code to show you the structure I mean:

```
foam.CLASS({
  name: 'VerticalMenuItemView',
  extends: 'foam.u2.View',

  properties: [
    {
      class: 'Boolean',
      name: 'expanded',
      value: false
    }
  ]

  methods: [
    function initE() {
      var view = this;
      
      this.
        add(this.data.label).
        on('click', function() {
          var menu = this.data;
          menu.children.select().then(function(c) {
            if ( c.array.length == 0 )
              menu.launch(view.__context__, view);
            else
              this.expanded = ! this.expanded;
          });
        }).
        add(this.slot(function(expanded, data) {
          return ! expanded ?
            this.E() :
            this.E().
            select(data.children, function(child) {
              return view.cls_.create({ data: child }, view);
            })
        }));
    }
  ]
});


foam.CLASS({
  name: 'VerticalMenuView',
  requires: [ 'VerticalMenuItemView' ],
  methods: [
    function initE() {
      var view = this;
      
      this.select(this.menuDAO, function(menu) {
        return this.VerticalMenuItemView.create({ data: menu });
      });
    }
  ]
});
```